x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -129,7 +129,12 @@
                     },
                     null, MetadataUtils.DEFAULT_RESERVED_METADATA))
                 .put(UsernamesField.BEATS_ROLE, new RoleDescriptor(UsernamesField.BEATS_ROLE,
-                        new String[] { "monitor", MonitoringBulkAction.NAME}, null, null, MetadataUtils.DEFAULT_RESERVED_METADATA))
+                        new String[] { "monitor", MonitoringBulkAction.NAME},
+                        new RoleDescriptor.IndicesPrivileges[]{
+                            RoleDescriptor.IndicesPrivileges.builder()
+                                .indices(".monitoring-beats-*").privileges("create_index", "write").build()
```

Comment:
Can we make this `create` instead of `write` ?
That gives the ability to index documents (including overwriting existing documents), but not call the update or delete APIs.

I believe that's sufficient for your purposes, and would get us closer to having _least privilege_ users & roles.

Reply:
Thanks, indeed `create` instead of `write` is sufficient here as the `.monitoring-beats-*` indices are meant to be append-only and the `create` privilege allows for bulk indexing as well.

Fixed in 6bf4d8b4756.